examples/shader/animate_shader.rs

```diff
@@ -112,8 +112,8 @@ fn queue_custom(
         | MeshPipelineKey::from_primitive_topology(PrimitiveTopology::TriangleList);
 
     for (view, mut transparent_phase) in views.iter_mut() {
-        let view_matrix = view.transform.compute_matrix();
-        let view_row_2 = view_matrix.row(2);
+        let inverse_view_matrix = view.transform.compute_matrix().inverse();
```

Comment:
I am not thrilled that users have to reason about this directly, but I'm not immediately sure if there's a clean abstraction to expose.

Reply:
Yeah, I was trying to think of a clear and efficient way of doing this. Calculating a matrix inverse is expensive. The matrices are column-major so extracting the row has a cost too. I don't know if there is much value to adding a function to calculate the inverse view matrix row 2. And then another for calculating the dot product of that with the translation/position of interest.

I suppose extracting row 2 isn't so expensive, so providing a couple of methods for calculating the view z from the inverse view and model transforms, and from the inverse view and a translation/point could be helpful. I don't know how we can make them discoverable though. Maybe we could solve it by making inlined constructor functions... something like:
```rust
impl Transparent3d {
    #[inline]
    pub fn from_mesh_transform(
        entity: Entity,
        pipeline: CachedRenderPipelineId,
        draw_function: DrawFunctionId,
        inverse_view: &Mat4,
        mesh_transform: &Mat4,
    ) -> Self {
        Self::from_mesh_translation(
            entity,
            pipeline,
            draw_function,
            inverse_view,
            mesh_transform.col(3),
        )
    }

    #[inline]
    pub fn from_mesh_translation(
        entity: Entity,
        pipeline: CachedRenderPipelineId,
        draw_function: DrawFunctionId,
        inverse_view: &Mat4,
        mesh_translation: Vec4,
    ) -> Self {
        Transparent3d {
            distance: inverse_view.row(2).dot(mesh_translation),
            pipeline,
            entity,
            draw_function,
        }
    }
}

Reply:
I realised the only place I had seen the Vec4 variant was for lights and that is internal. Everything else used the full transform. Check it out! :)

Also, I did a quick test to verify that this didn't have any impact on performance of adding Transparent2d phase items to the Transparent2d RenderPhase for sprites due to using a constructor function. I added `#[inline]` annotations but wanted to be sure. `many_sprites` `queue_sprites` takes 11.39ms on `main` and 11.47ms on this PR. 0.7% slower. Probably just noise...